src/Nerdbank.MessagePack/MessagePackPrimitiveSpanUtility.cs

```diff
@@ -0,0 +1,1831 @@
+﻿#pragma warning disable SA1636 // File header copyright text should match
+// Copyright (c) All Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#pragma warning restore SA1636 // File header copyright text should match
+
+using System.Buffers.Binary;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+
+namespace Nerdbank.MessagePack;
+
+/// <summary>
+/// Read/Write methods with Hardware Intrinsics for primitive msgpack codes.
+/// </summary>
+internal static class MessagePackPrimitiveSpanUtility
+{
+	/// <summary>
+	/// Read <see cref="bool"/> values.
+	/// </summary>
+	/// <param name="output">The reference to the bool buffer which receives the values.</param>
+	/// <param name="input">The reference to the source byte buffer.</param>
+	/// <param name="length">The length of the input/output buffers.</param>
+	/// <returns><see langword="true" /> if <paramref name="input"/> were all valid; otherwise, <see langword="false" />.</returns>
+	public static bool Read(ref bool output, ref byte input, nuint length)
```

Comment:
In this case, the input length can be trivially predicted because a msgpack bool value is always exactly 1 byte. But how do you deal with variable-length elements such as integers? If we can't predict how long the input buffer is before starting the read, we can't ensure they are contiguous in memory (since we deserialize from a `ReadOnlySequence<byte>`).

Reply:
I guess the answer to this is you can't. At least, [it appears you don't](https://github.com/MessagePack-CSharp/MessagePack-CSharp/blob/e6f3d81f96578595c675966747dfdd5cac26528a/src/MessagePack.Experimental/HardwareIntrinsics/FormatterHelpers.cs#L305-L325).

Reply:
No I can't.